### DISCUSSION TITLE
DBeaver shut down unexpectedly and will not reopen

### DISCUSSION THREAD

**Discussion by aynmo22** [dbeaver/dbeaver#14574] [General]
#### System information: 
- Operating system (distribution) and version
- DBeaver version
- Additional extensions

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Include any warning/errors/backtraces from the logs

he app shut down unexpectedly and despite how many times I click 'OK' to acknowledge and move forward, DBeaver does not reopen. I cannot access any aspects of the application outside of the error menu which allows me to 'Hide Details' or 'OK'. 

Process:               dbeaver [2128]
Path:                  /Applications/DBeaver.app/Contents/MacOS/dbeaver
Identifier:            org.jkiss.dbeaver.core.product
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           dbeaver [2128]
User ID:               501

Date/Time:             2021-10-18 17:42:23.271 -0700
OS Version:            Mac OS X 10.15.7 (19H2)
Report Version:        12
Bridge OS Version:     4.6 (17P6610)
Anonymous UUID:        377E5A8F-4DA5-D84C-0C69-4F69A391A5D6

Sleep/Wake UUID:       7ED44B49-D18F-4F23-A3C8-4CDE0FFDC904

Time Awake Since Boot: 8100 seconds
Time Since Wake:       3800 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
-->
    __TEXT                 0000000008fc7000-0000000008fcb000 [   16K] r-x/r-x SM=COW 

Thread 0 Crashed:
0   ???                                 0x00000000134be000 _dyld_start + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffee6c3cc48
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x00000000134be000  rfl: 0x0000000000000200  cr2: 0x0000000000000000

Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0

Binary Images:
         0x8fc7000 -          0x8fcaff7 +??? (0) <BF0BAA93-FE7A-3039-AC22-E0007DC821F4> (null)
        0x134bd000 -         0x1354ef47 +??? (750.6) <1D318D60-C9B0-3511-BE9C-82AFD2EF930D> (null)

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 5829
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=852K resident=0K(0%) swapped_out_or_unallocated=852K(100%)
Writable regions: Total=8404K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8404K(100%)

                                VIRTUAL   REGION
REGION TYPE                        SIZE    COUNT (non-coalesced)
===========                     =======  =======
STACK GUARD                       56.0M        1
Stack                             8192K        1
__DATA                             232K        3
__DATA_CONST                        20K        1
__LINKEDIT                         252K        2
__TEXT                             600K        2
shared memory                        8K        2
===========                     =======  =======
TOTAL                             65.1M       12

Model: MacBookPro16,1, BootROM 1037.147.4.0.0 (iBridge: 17.16.16610.0.0,0), 6 processors, 6-Core Intel Core i7, 2.6 GHz, 16 GB, SMC
Graphics: kHW_IntelUHDGraphics630Item, Intel UHD Graphics 630, spdisplays_builtin
Graphics: kHW_AMDRadeonPro5300MItem, AMD Radeon Pro 5300M, spdisplays_pcie_device, 4 GB
Memory Module: BANK 0/ChannelA-DIMM0, 8 GB, DDR4, 2667 MHz, Micron, 8ATF1G64HZ-2G6E1
Memory Module: BANK 2/ChannelB-DIMM0, 8 GB, DDR4, 2667 MHz, Micron, 8ATF1G64HZ-2G6E1
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x7BF), wl0: Apr  7 2020 13:09:12 version 9.30.357.41.32.5.49 FWID 01-617711e7
Bluetooth: Version 7.0.6f7, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB3.1 Hub
USB Device: USB 3.1 Bus
USB Device: USB2.1 Hub
USB Device: Magic Keyboard with Numeric Keypad
USB Device: 2.4G Wireless Mouse
USB Device: Apple T2 Bus
USB Device: Composite Device
USB Device: Touch Bar Backlight
USB Device: Touch Bar Display
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Headset
USB Device: Ambient Light Sensor
USB Device: FaceTime HD Camera (Built-in)
USB Device: Apple T2 Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 55.3
Thunderbolt Bus: MacBook Pro, Apple Inc., 55.3

**Comment by uslss** [comment]
@aynmo22 please provide steps to reproduce the issue. Video with the issue also will be very helpful

**Comment by andrelinslima** [comment]
"Termination Reason: Namespace CODESIGNING, Code 0x1" means Mac OS X is complaining about the app signing not matching the expected or other OS related restriction.
Did you install the application from the AppStore (assuming it's available there) ? Did you upgrade the application before the problem? What is dbeaver version currently? 
If you updated Dbeaver from outside the App Store, make sure your configuration under Security & Privacy -> "Allow apps downloaded from" is set to "App Store and identified developers" instead of only App Store.
